APCAppCore/APCAppCore/Library/Categories/NewDataCollector/Collectors/APCDisplacementTrackingCollector.m

```diff
@@ -0,0 +1,175 @@
+    
+    self = [super initWithIdentifier:identifier dateAnchorName:nil launchDateAnchor:nil];
+    
+    if (self != nil)
+    {
+        _deferringUpdates = NO;
+        
+        [self setupInitialLocationParameters];
+    }
+    
+    return self;
+}
+
+- (void)dealloc
+{
+    [[NSNotificationCenter defaultCenter] removeObserver:self];
+}
+
+- (void)setupInitialLocationParameters
+{
+    _baseTrackingLocation       = [[CLLocation alloc] initWithLatitude:0.0 longitude:0.0];
+    _mostRecentUpdatedLocation  = [[CLLocation alloc] initWithLatitude:0.0 longitude:0.0];
+}
+
+- (void)start
+{
+    if ([CLLocationManager locationServicesEnabled])
+    {
+        if (!self.locationManager)
+        {
+            APCLogDebug(@"Start location tracking");
+            
+            self.locationManager            = [[CLLocationManager alloc] init];
+            self.locationManager.delegate   = self;
+            
+            if ([CLLocationManager significantLocationChangeMonitoringAvailable] &&
+                [CLLocationManager authorizationStatus] == kCLAuthorizationStatusAuthorizedAlways)
+            {
+                APCLogDebug(@"Significant Location Change Monitoring is Available");
+                
+                [self.locationManager startMonitoringSignificantLocationChanges];
+            }
+        }
+    }
+}
+
+- (void)stop
+{
+    if ([CLLocationManager locationServicesEnabled])
+    {
+        if ([CLLocationManager significantLocationChangeMonitoringAvailable])
+        {
+            [self.locationManager stopMonitoringSignificantLocationChanges];
+        }
+    }
+}
+
+/*********************************************************************************/
+#pragma mark - Private Methods
+/*********************************************************************************/
+
+- (void)updateArchiveDataWithLocationManager:(CLLocationManager*)manager withUpdateLocations:(NSArray*)locations
+{
+    //Send to delegate
+    if ([self.delegate respondsToSelector:@selector(didReceiveUpdateWithLocationManager:withUpdateLocations:)])
+    {
+        [self.delegate didReceiveUpdateWithLocationManager:manager withUpdateLocations:locations];
+    }
+}
+
+/*********************************************************************************/
+#pragma mark -CLLocationManagerDelegate
+/*********************************************************************************/
+
+- (void)locationManager:(CLLocationManager*)manager didFailWithError:(NSError*)error
+{
+    APCLogError2(error);
+    
+    switch(error.code)
+    {
+        case kCLErrorNetwork:
+        {
+            //  Possible network connection issue (eg, in airplane mode)
+            break;
+        }
+            
+        case kCLErrorDenied:
+        {
+            //  The user has denied use of location
+            [manager stopUpdatingLocation];
+            //  The app delegate should be notified so that other components (eg, Profile) could
```

Comment:
Wait - this code doesn't actually do anything to notify the app delegate or anyone else. Where does the error go?